features/migration/impl/src/main/kotlin/io/element/android/features/migration/impl/MigrationPresenter.kt

```diff
@@ -24,40 +24,46 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import io.element.android.features.api.MigrationState
-import io.element.android.features.rageshake.api.logs.LogFilesRemover
+import io.element.android.features.migration.impl.migrations.AppMigration
 import io.element.android.libraries.architecture.AsyncData
 import io.element.android.libraries.architecture.Presenter
+import io.element.android.libraries.di.AppScope
+import io.element.android.libraries.di.SingleIn
+import timber.log.Timber
 import javax.inject.Inject
 
+@SingleIn(AppScope::class)
 class MigrationPresenter @Inject constructor(
     private val migrationStore: MigrationStore,
-    private val logFilesRemover: LogFilesRemover,
+    private val migrations: Set<@JvmSuppressWildcards AppMigration>,
 ) : Presenter<MigrationState> {
     @Composable
     override fun present(): MigrationState {
-        val migrationStoreVersion = migrationStore.applicationMigrationVersion().collectAsState(initial = null)
+        val migrationStoreVersion by migrationStore.applicationMigrationVersion().collectAsState(initial = null)
         var migrationAction: AsyncData<Unit> by remember { mutableStateOf(AsyncData.Uninitialized) }
 
-        /*
         // Uncomment this block to run the migration everytime
-        LaunchedEffect(Unit) {
-            migrationStore.setApplicationMigrationVersion(0)
-        }
-         */
+//        LaunchedEffect(Unit) {
+//            Timber.d("Resetting migration version to 0")
+//            migrationStore.setApplicationMigrationVersion(0)
+//        }
+
+        val orderedMigrations = migrations.sortedBy { it.order }
```

Comment:
This could be moved as a Class member I think, and so `migrations` should not be a class member, to avoid any mistake.

Reply:
You mean storing it as a property and making `migrations` just a constructor parameter?